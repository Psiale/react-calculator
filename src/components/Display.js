import PropTypes from 'prop-types';
import React from 'react';

const Display = ({ result }) => (
  <>
    <div className="result">{result}</div>
  </>
);

Display.propTypes = {
  result: PropTypes.string,
};

Display.defaultProps = {
  result: '0',
};
export default Display;
