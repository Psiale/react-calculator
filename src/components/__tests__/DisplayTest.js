import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import renderer from 'react-test-renderer';
import Display from '../Display';

configure({ adapter: new Adapter() });

describe("Describe the display component", () => {
    it('matches the snapshot', () => {
        const tree = renderer.create(<Display result="0" />).toJSON;
        expect(tree).toMatchSnapshot();
      });
});
