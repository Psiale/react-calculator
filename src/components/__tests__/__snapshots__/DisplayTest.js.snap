// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Describe the display component matches the snapshot 1`] = `[Function]`;
